satpy/tests/reader_tests/test_ici_l1b_nc.py

```diff
@@ -0,0 +1,432 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+#
+# Copyright (c) 2020 Satpy developers
+#
+# satpy is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# satpy is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with satpy.  If not, see <http://www.gnu.org/licenses/>.
+"""The ici_l1b_nc reader tests package.
+
+This version tests the reader for ICI test data as per PFS V3A.
+
+"""
+
+import os
+import unittest
+import uuid
+from datetime import datetime
+from unittest.mock import ANY, patch
+
+import numpy as np
+import pytest
+import xarray as xr
+from netCDF4 import Dataset
+
+from satpy.readers.ici_l1b_nc import IciL1bNCFileHandler
+
+TEST_FILE = 'test_file_ici_l1b_nc.nc'
+
+
+class TestIciL1bNCFileHandler(unittest.TestCase):
```

Comment:
we're trying to move away from unittest and use pytest, so this class doesn't need to inherit from TestCase.
In that case, you can use the `tmp_path` fixture and remove the teardown.

Reply:
refactored so no using pytest